engine/common/requester/engine.go

```diff
@@ -426,6 +426,14 @@ func (e *Engine) onEntityResponse(originID flow.Identifier, res *messages.Entity
 		delete(needed, entityID)
 		delete(e.items, entityID)
 
+		actualEntityID := entity.ID()
+
+		// validate that we got correct entity, exactly what we were expecting
+		if entityID != actualEntityID {
```

Comment:
I think this check might break the consensus node. 

The entity request is currently used not only for requesting an entity, but also the data associated with a certain key. 

For instance, the consensus matching engine will [request pending receipts by block ID](https://github.com/onflow/flow-go/blob/master/engine/consensus/matching/engine.go#L949), so it passes the block ID to the requester to fetch the receipts for that block. So here the entity being requested is the receipt, but we are not requesting it with the receipt ID, instead, with block ID. 

So when we received the requested receipt, its ID will be different from the block ID, and then we would reject it.

This check can not be applied to all the cases, but only the case where the requester is to request entitys by its ID